owner-onboarding-server/src/main.rs

```diff
@@ -346,6 +415,48 @@ async fn main() -> Result<()> {
             .context("Error converting owner public key to PK")?
     };
 
+    // Voucher registration times
+    let str_registration_period = settings.registration_period;
+    let str_re_registration_window = settings.re_registration_window;
+    let registration_period = match str_registration_period {
+        Some(value) => {
+            let tmp = value
+                .parse::<u32>()
```

Comment:
If you make the type `Option<u32>`, you don't have to parse and handle parse errors anymore.

Reply:
Related to the first comment, in this way I check that the number is not negative while I parse it.

Reply:
Yep, fair enough then.